src/mdbExtensionController.ts

```diff
@@ -397,35 +397,71 @@ export default class MDBExtensionController implements vscode.Disposable {
     );
   }
 
-  public openMongoDBShell(): Promise<boolean> {
-    let mdbConnectionString;
-
-    if (this._connectionController) {
-      const activeConnectionModel = this._connectionController
-        .getActiveConnectionModel()
-        ?.getAttributes({ derived: true });
+  isSslConnection(activeConnectionModel: any): boolean {
+    return !!(
+      activeConnectionModel &&
+      activeConnectionModel.driverOptions &&
+      (activeConnectionModel.driverOptions.sslCA ||
+        activeConnectionModel.driverOptions.sslCert ||
+        activeConnectionModel.driverOptions.sslPass)
+    );
+  }
 
-      mdbConnectionString = activeConnectionModel
-        ? activeConnectionModel.driverUrlWithSsh
-        : '';
-    }
+  public openMongoDBShell(): Promise<boolean> {
+    let mdbConnectionString: any;
+    let mdbTlsOptionsString = '';
 
-    if (!mdbConnectionString) {
+    if (!this._connectionController) {
       vscode.window.showErrorMessage(
         'You need to be connected before launching the MongoDB Shell.'
       );
 
       return Promise.resolve(false);
     }
 
+    const activeConnectionModel = this._connectionController
+      .getActiveConnectionModel()
+      ?.getAttributes({ derived: true });
+
+    mdbConnectionString = activeConnectionModel
+      ? activeConnectionModel.driverUrlWithSsh
+      : '';
+
     const mongoDBShell = vscode.window.createTerminal({
       name: 'MongoDB Shell',
       env: { MDB_CONNECTION_STRING: mdbConnectionString }
     });
+
+    if (this.isSslConnection(activeConnectionModel)) {
+      mdbTlsOptionsString = '--tls';
+
+      if (!activeConnectionModel?.driverOptions.checkServerIdentity) {
+        mdbTlsOptionsString = `${mdbTlsOptionsString} --tlsAllowInvalidHostnames`;
+      }
+
+      if (!activeConnectionModel?.driverOptions.sslValidate) {
+        mdbTlsOptionsString = `${mdbTlsOptionsString} --tlsAllowInvalidCertificates`;
+      }
+
+      if (activeConnectionModel?.driverOptions.sslCA) {
+        mdbTlsOptionsString = `${mdbTlsOptionsString} --tlsCAFile ${activeConnectionModel?.driverOptions.sslCA}`;
+      }
+
+      if (activeConnectionModel?.driverOptions.sslCert) {
+        mdbTlsOptionsString = `${mdbTlsOptionsString} --tlsCertificateKeyFile ${activeConnectionModel?.driverOptions.sslCert}`;
+      }
+
+      if (activeConnectionModel?.driverOptions.sslPass) {
+        mongoDBShell['MDB_TLS_CERTIFICATE_KEY_FILE_PASSWORD'] =
```

Comment:
Ah didn't know environment variables could be set like this. 
Maybe we move the forming of `mdbTlsOptionsString` into a helper to make this function a little smaller?

Reply:
Yes also was thinking about it!

Reply:
And you can not set env like this, used worn variable name. Thanks for noticing!